--emacs flag doesn't include the rule's message, only code snippet

**Describe the bug**

The `--emacs` flag doesn't include the human readable `message` for failing rules in its output, which makes it harder to work out what the appropriate fix is.

(Thanks for semgrep; we're using it for both security analysis and also a few custom rules to catch (non-security) mistakes/typos we keep making. Works well!)

**To Reproduce**

Run `semgrep` with the `--emacs` flag. For instance, in the semgrep-rules repo:

```shell
semgrep --emacs -c yaml/semgrep/missing-message-field.yaml yaml/semgrep/missing-message-field.test.yaml
```
```
running 1 rules...
yaml/semgrep/missing-message-field.test.yaml:3:3:warning(missing-message-field):- id: unchecked-subprocess-call
ran 1 rules on 1 files: 1 findings
```

This is showing the relevant code snippet (`- id: unchecked-subprocess-call`), but no human readable context.

**Expected behavior**

The output should include the human readable context included in the `message` field, similar to the `--vim` flag:

```shell
semgrep --vim -c yaml/semgrep/missing-message-field.yaml yaml/semgrep/missing-message-field.test.yaml
```
```
running 1 rules...
yaml/semgrep/missing-message-field.test.yaml:3:3:W:yaml.semgrep.missing-message-field:This rule does not have a message. Semgrep requires that
rules have a message. Include a message to explain what the rule does.
Consider writing a message that explains why this is an issue and
how to fix it.

ran 1 rules on 1 files: 1 findings
```

(I've filed a separate https://github.com/returntocorp/semgrep-rules/issues/1431 about the message being broken across multiple lines even with the nominally single-line `--vim` output format.)

**Screenshots**
N/A

**What is the priority of the bug to you?**

- [ ] P0: blocking your adoption of Semgrep or workflow
- [ ] P1: important to fix or quite annoying
- [x] P2: regular bug that should get fixed

**Environment**
macOS, semgrep installed from homebrew, version 0.61.0

Thanks for digging up some history; that doesn't seem to discuss why it's missing the rule's message specifically. I don't think that's required by emacs' error message matching, for instance, here's a little experiment with GCC, which emacs is happy to consume directly in its compilation-mode:

```
$ cat foo.c
i wrote some bad syntax here
$ gcc foo.c
foo.c:1:1: error: unknown type name 'i'
    1 | i wrote some bad syntax here
      | ^
foo.c:1:9: error: expected '=', ',', ';', 'asm' or '__attribute__' before 'some'
    1 | i wrote some bad syntax here
      |         ^~~~
foo.c:1:9: error: unknown type name 'some'
```

That is, the description is included on the 'main' line, and even code snippets.

--- a/semgrep/semgrep/formatter/emacs.py
+++ b/semgrep/semgrep/formatter/emacs.py
@@ -19,6 +19,7 @@ class EmacsFormatter(BaseFormatter):
             str(rule_match.start["col"]),
             severity,
             rule_match.lines[0].rstrip(),
+            rule_match.message,
         ]
 
     def output(self) -> str:

--- a/semgrep/tests/e2e/snapshots/test_check/test_registry_rule/results.json
+++ b/semgrep/tests/e2e/snapshots/test_check/test_registry_rule/results.json
@@ -10,7 +10,7 @@
       "extra": {
         "is_ignored": false,
         "lines": "console.log(x == x)",
-        "message": "Detected a useless comparison operation `x == x` or `x != x`. This\noperation is always true.\nIf testing for floating point NaN, use `math.isnan`, or\n`cmath.isnan` if the number is complex.\n",
+        "message": "Detected a useless comparison operation `x == x` or `x != x`. This operation is always true. If testing for floating point NaN, use `math.isnan`, or `cmath.isnan` if the number is complex.",
         "metadata": {
           "category": "correctness",
           "license": "Commons Clause License Condition v1.0[LGPL-2.1-only]",

--- a/semgrep/tests/e2e/snapshots/test_output/test_output_format/--emacs/results.out
+++ b/semgrep/tests/e2e/snapshots/test_output/test_output_format/--emacs/results.out
@@ -1,1 +1,1 @@
-targets/basic/stupid.py:3:12:error(eqeq-is-bad):    return a + b == a + b
+targets/basic/stupid.py:3:12:error(eqeq-is-bad):    return a + b == a + b:useless comparison operation `a + b == a + b` or `a + b != a + b`; possible bug?

